src/api/app/controllers/webui/project_controller.rb

```diff
@@ -56,24 +56,24 @@ def index
   end
 
   def autocomplete_projects
-    render json: Project.autocomplete(params[:term]).not_maintenance_incident.order(:name).pluck(:name)
+    render json: Project.autocomplete(params[:term]).not_maintenance_incident.order(:name).limit(50).pluck(:name)
   end
 
   def autocomplete_incidents
-    render json: Project.autocomplete(params[:term]).maintenance_incident.order(:name).pluck(:name)
+    render json: Project.autocomplete(params[:term]).maintenance_incident.order(:name).limit(50).pluck(:name)
   end
 
   def autocomplete_packages
     @project = Project.find_by(name: params[:project])
     if @project
-      render json: @project.packages.autocomplete(params[:term]).pluck(:name)
+      render json: @project.packages.autocomplete(params[:term]).limit(50).pluck(:name)
     else
       render json: nil
     end
   end
 
   def autocomplete_repositories
-    render json: @project.repositories.pluck(:name)
+    render json: @project.repositories.limit(50).pluck(:name)
```

Comment:
That means there are some repositories you can not select :see_no_evil:  (at least in the repository page). You need some criteria here.

Also, how likely is that we have more than 50 repositories in a project? :eyes: